storage/id_alloc.go

```diff
@@ -58,10 +58,10 @@ type idAllocator struct {
 func newIDAllocator(idKey proto.Key, db *client.DB, minID uint32, blockSize uint32, stopper *stop.Stopper) (*idAllocator, error) {
 	// minID can't be the zero value because reads from closed channels return
 	// the zero value.
-	if minID == 0 {
+	if minID <= 0 {
```

Comment:
they're both unsigned ints, the only thing they can do is overflow before this call and be a really massive number.